llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp

```diff
@@ -521,14 +522,16 @@ void CombineRuleOperandTypeChecker::getInstEqClasses(
   //    - Iterating over the map, filtering types we don't like, and just adding
   //      the array of Operand Indexes to \p OutTECs.
 
-  // We can only do this on CodeGenInstructions. Other InstructionPatterns have
-  // no type inference information associated with them.
+  // We can only do this on CodeGenInstructions that aren't intrinsics. Other
+  // InstructionPatterns have no type inference information associated with
+  // them. Intrinsics also don't have any constraints to guide inference.
```

Comment:
This isn't entirely true? The signatures do support some level of related parameter constraints